service/src/main/java/com/appirio/service/billingaccount/dao/BillingAccountDAO.java

```diff
@@ -388,4 +388,63 @@ void updateChallengeFeePercentage(
      */
     @SqlUpdateFile("sql/billing-account/challenge-fees/delete-challenge-fee.sql")
     void deleteChallengeFee(@ApiQueryInput QueryParameter queryParameter);
+
+    /**
+     * Count number of entries for a Challenge Id in a Project
+     *
+     * @param projectId : Refers to the Billing Account ID
+     * @param challengeId : Refers to the Challenge ID
+     * 
+     */
+    @SqlQueryFile("sql/billing-account/budget-amount/count-challengeid-entries.sql")
+    Long countChallengeIdEntries(@Bind("projectId") long projectId, @Bind("challengeId") String challengeId);
```

Comment:
The DAO does not support return a Long. It throws exception:
`java.lang.IllegalArgumentException: A bean, java.lang.Long, was mapped which was not instantiable`

Need use IdDTO as existing code.